### PR TITLE
Setup Person, Student and Teacher Classes

### PR DESCRIPTION
**Pull request by aamir-asaram**
**In this PR, I have implemented the following:**
- Created a super class `Person` as well as subclasses `Student` and `Teacher` that extend the `Person` class.

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './Person'
+
+class Student < Person
+  def initialize(classroom)
+    super(name, age, parents_permission)
+    @classroom = classroom
+  end
```
Comment (TedLivist) [inline student.rb]: - [ ] Kindly fix the initialize method to request `age`, `name` and `parents_permission`.

- [ ] Also the `#super` method arguments are missing the correct order.

  ```
  def initialize(age, classroom, parents_permission: true, name: 'Unknown')
    super(age, parents_permission: parents_permission, name: name)
    @classroom = classroom
  end
  ```

File: teacher.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require './Person'
+
+class Teacher < Person
+  def initialize(specialization)
+    super(name, age)
+    @specialization = specialization
+  end
```
Comment (TedLivist) [inline teacher.rb]: - [ ] Kindly fix the initialize method to request `age`, `name` and `parents_permission`.

- [ ] Also the `#super` method arguments are missing the correct order.

  ```
  def initialize(age, specialization, parents_permission: true, name: 'Unknown')
    super(age, parents_permission: parents_permission, name: name)
    @specialization = specialization
  end
  ```

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @aamir-asaram,

Good job so far!
✔️ Classes are in different files based on convention
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: teacher.rb (hunk 2 above)
Comment (aamir-asaram) [reply teacher.rb]: Thank you for the valuable feedback! @TedLivist  :smile:

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @aamir-asaram👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with classes. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**